### DISCUSSION TITLE
Data Provider: Filtering binding to custom Current User field not working

### DISCUSSION THREAD

**Discussion by reichert** [Budibase/budibase#5029] [Help]
Hi guys, this is my first post here!

What I did:
1) I added a column called 'tenant' in the default users table;
2) I created a model using the internal database with a column called 'tenant' as well;
3) In the "list view" of the app I added a filter to the datasource as shown below:
![image](https://user-images.githubusercontent.com/1287032/159592042-7ba25cac-450c-4b86-b174-88fc8639a56c.png)

4) I updated in my user a value in the tenant column like below:
![image](https://user-images.githubusercontent.com/1287032/159592208-36d1f346-d681-42dc-9b38-4cb8286503d9.png)

5) but when I run the app the rows are not being filtered by this column. More than that, in the inspector it seams that the {{ Current User.tenant }} is not being evaluated:
![image](https://user-images.githubusercontent.com/1287032/159592758-778cb585-60e1-4c5e-be7e-d1d72f63d163.png)

Any suggestion?

**Comment by melohagan** [comment]
Hi @reichert, thanks for your interest in Budibase!

Could you have a look at the **self** tab under the network inspector to verify that the user you expect is logged in:

![Screenshot 2022-03-23 at 10 30 06](https://user-images.githubusercontent.com/101575380/159680095-ddd709f9-fb2f-41b2-9313-69c82cacf706.png)

If the tenant filter is an empty string, then all rows will be returned.

**Comment by reichert** [comment]
Helo Melohagan, thanks for your quick response.
Yes the user is the expected one and as you can see the tenants are different:

![image](https://user-images.githubusercontent.com/1287032/159695123-621271db-82b7-4366-950b-9f23fd335a67.png)

The app is also published and just confirming - here is the filter I configured:

![image](https://user-images.githubusercontent.com/1287032/159695366-992b6726-dfe0-4499-835a-405ffd740d56.png)

Also checked for typos and everything seems to be fine. The field name in the users table is lowercase.

Anything else I should check?

**Reply by melohagan** [reply]
The filters should apply to the Data Provider within the builder. So when you click save on the filter a green toast should appear stating **Filters saved**.
Then click off the Data Provider to get it to refresh and display the filtered results.

Is the filtering working within the builder?

**Comment by reichert** [comment]
Hi, checking here it seams the editor is not saving when changing from "Value" to "Binding"
![image](https://user-images.githubusercontent.com/1287032/159913734-8838f608-3ac5-48c1-8fdf-72c9f68f22d6.png)

After removing the filter and adding again it worked fine.
I thing this should be addressed as a simple bug to the development team.

Have a nice day!

**Reply by melohagan** [reply]
I'm glad you got it sorted, and I hope you have a nice day too! :)

I'm not able to recreate the bug myself, but if you were able to send over any errors from the network tab that would be very helpful. Otherwise I'll keep an eye out for it in the future.

**Comment by reichert** [comment]
Hi, here are the steps to reproduce the bug:
0) create a new project using an internal table and create all the CRUD screens automatically. Then edit the list view following the next steps:
1) create a filter based on VALUE and save
2) edit this filter to be based on BINDING and save again
3) the screen will not obey this filter based on BINDING

I checked that:
1) the post to /screens correctly send the parameter BINDING to update the filter:
![image](https://user-images.githubusercontent.com/1287032/159956591-0c270557-68df-4bbb-8b37-85e5164751fd.png)
2) the get of the entire page again (after a refresh) correctly get the filter as BINDING
![image](https://user-images.githubusercontent.com/1287032/159956758-be9a30b7-5efe-496f-8ca9-5029a5940ceb.png)
3) however the filter is not applied

Workaround:
- remove the filter and create it again

Thinkgs to check:
- I guess there may be a second table where this filter is updated and the update of the filter through post is just updated in one table.

**Reply by reichert** [reply]
Hello @melohagan tell me if you need any aditional help to reproduce the error, ok?
It is not affecting me, but it can cause a buggy image to other users - something in this case easy to avoid :)

Best wishes!

**Reply by melohagan** [reply]
@reichert This was very helpful, thank you! I've raised a detailed bug here: https://github.com/Budibase/budibase/issues/5105

I noticed another buggy scenario related to this, so it'll be great to get sorted!
